arbeitszeit_web/presenters/end_cooperation_presenter.py

```diff
@@ -35,9 +35,10 @@ def present(self, response: EndCooperationResponse) -> ViewModel:
         return self.ViewModel(show_404=False, redirect_url=redirect_url)
 
     def _get_redirect_url(self) -> str:
-        referer = self.request.get_environ("HTTP_REFERER")
-        plan_id = self.request.get_arg("plan_id")
-        cooperation_id = self.request.get_arg("cooperation_id")
+        referer = self.request.get_header("HTTP_REFERER")
```

Comment:
It must be changed to `self.request.get_header("Referer")`.
Right now it always returns None, i.e. it always redirects to the cooperation page.